coderd/coderd.go

```diff
@@ -135,7 +135,13 @@ type Options struct {
 	AccessControlStore          *atomic.Pointer[dbauthz.AccessControlStore]
 	// AppSecurityKey is the crypto key used to sign and encrypt tokens related to
 	// workspace applications. It consists of both a signing and encryption key.
-	AppSecurityKey     workspaceapps.SecurityKey
+	AppSecurityKey workspaceapps.SecurityKey
+
+	// The following two functions are dependencies of HealthcheckFunc but are only implemented
+	// in enterprise. Stubbing them out here.
+	FetchWorkspaceProxiesFunc *atomic.Pointer[func(context.Context) (codersdk.RegionsResponse[codersdk.WorkspaceProxy], error)]
+	UpdateProxyHealthFunc     *atomic.Pointer[func(context.Context) error]
```

Comment:
Suggestion: How would you feel about, instead of multiple pointer funcs, we would use a service instead?

```go
type WorkspaceProxies interface {
	Fetch()
	UpdateHealth()
}
```

Seems convenient to pass the service around instead of multiple funcs, also beneficial if this is ever extended to add more funcs.

Reply:
I considered that approach initially! This just ended up being less code overall:

We'd have to:
- Make the AGPL no-op implementation of the service
- Make the enterprise implementation of the service
- Refactor the current way the enterprise API fetches workspace proxies etc.
- Plumb all of those things through, wire them up, etc.

For the sake of just two funcs, I considered all that a bit overkill.

Reply:
I'll give you points 1 and 2, but it seems like it would still be very little code to have those.

> * Refactor the current way the enterprise API fetches workspace proxies etc.
> * Plumb all of those things through, wire them up, etc.

Not sure I agree that there's a lot of refactoring/plumbing needed? Consider the following:

```go
// AGPL
type WorkspaceProxyNoop struct {}

func (WorkspaceProxyNoop) FetchWorkspaceProxies(context.Context) (codersdk.RegionsResponse[codersdk.WorkspaceProxy], error) {
	return codersdk.RegionsResponse[codersdk.WorkspaceProxy]{}, nil
}

func (WorkspaceProxyNoop) UpdateProxyHealth(context.Context) error {
	return nil
}

api.AGPL.WorkspaceProxy.Store(WorkspaceProxyNoop{})

// Enterprise
type WorkspaceProxy struct {
	FetchWorkspaceProxies (context.Context) (codersdk.RegionsResponse[codersdk.WorkspaceProxy], error)
	UpdateProxyHealth (context.Context) error
}

api.AGPL.WorkspaceProxy.Store(WorkspaceProxy{
	FetchWorkspaceProxies: api.fetchWorkspaceProxies
	UpdateProxyHealth: api.ProxyHealth.ForceUpdate
})
```

Wouldn't that cover it without additional plumbing or refactoring?

Reply:
Refactored.